src/Microsoft.Azure.WebJobs.Host/Extensions/ITriggerDataExtensions.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host.Bindings;
+using Microsoft.Azure.WebJobs.Host.Triggers;
+
+namespace Microsoft.Azure.WebJobs.Host
+{
+    internal static class ITriggerDataExtensions
+    {
+        public static Task<IReadOnlyDictionary<string, object>> GetBindingDataAsync(this ITriggerData triggerData)
+        {
+            if (triggerData is IBindingDataFactory bindingDataFactory)
+            {
+                return bindingDataFactory.GetBindingDataAsync();
+            }
+            else
+            {
+                return Task.FromResult(triggerData.BindingData);
+            }
+        }
+
+        public static IValueBinder GetReturnValueBinderOrNull(this ITriggerData triggerData)
+        {
+            // TODO: we should NOT be casting to a specific type like this, since the binding interfaces are based
```

Comment:
In the next major version, we need to clean up the ITriggerData/TriggerData APIs